### PR TITLE
[merged] Atomic/pull.py: Fix logic error in pull

### PR DESCRIPTION
**Pull request by baude**
When a image without a registry is provided as input, i.e.

atomic pull busybox

We are not able to resolve that reference now because we can
no longer depend on docker's algo's for that as we now
use skopeo. Therefore, we now take the input, see if skopeo
inspect can resolve it; and then decompose the fqdn so we
can check if the registry is secure.  Had to make a small
change on the return types of util.decompose to ensure
it returns str objects or else it will return unicode in
python2.

### CODE REVIEW

**Comment by baude** [comment]
rebased... @rhatdan please review ... no real code changes, just logic.

**Comment by baude** [comment]
@rhatdan hold on this.

**Comment by baude** [comment]
ok should be good.

File: Atomic/pull.py [hunk 1]
```diff
@@ -27,15 +27,16 @@ def cli(subparser):
 class Pull(Atomic):
     def pull_docker_image(self):
         registry, _, tag = decompose(self.args.image)
```
Comment (rhatdan) [inline Atomic/pull.py]: No longer use this registry, should probably be _

File: Atomic/util.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def decompose(compound_name):
         reg, repo = repo.split('/', 1)
     if ':' in repo:
         repo, tag = repo.rsplit(':', 1)
-    return reg, repo, tag
+    return str(reg), str(repo), str(tag)
```
Comment (rhatdan) [inline Atomic/util.py]: Why is this needed and will it work on both python2 and python3

Comment (baude) [inline Atomic/util.py]: in python2,  the skopeo outputs are all unicode.

File: Atomic/pull.py (hunk 1 above)
Comment (baude) [inline Atomic/pull.py]: indeed, will fix that.

File: Atomic/util.py (hunk 2 above)
Comment (baude) [inline Atomic/util.py]: prefer i type convert this prior to calling decompose?

Comment (rhatdan) [inline Atomic/util.py]: No I just wanted to make sure you tested it on both pythons.  Code like this can sometimes trigger other issues.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+
